docs/QA_MIGRATIONS_GUIDE.md

```diff
@@ -0,0 +1,17 @@
+### QA Migrations Guide
+Migrations are needed to change top-level state data, this can be found in the browser's storage. This can look like removing specific keys/value pairs from state, changing objects to an array of objects, changing the name of a controller, etc.
+
+Steps
+  1. Create a new MetaMask build\* folder locally with the source files before the migration, and load it as an unpacked extension in Chrome\*. If the migration is in a pull request, then build the `develop` branch to load. If the migration is already in `develop`, get a commit before the migration was added to build.
```

Comment:
Nit but do we want `build/` for consistent directory references?  Do we also want to use the term `directory`?